cpp/src/arrow/python/python_to_arrow.cc

```diff
@@ -738,6 +748,8 @@ class PyListConverter : public ListConverter<T, PyConverter, PyConverterTrait> {
           RETURN_NOT_OK(value_builder->Append(values[i]));
         }
       }
+    } else if (!values.is_strided()) {
```

Comment:
This is unrelated to the fix, but quality of life improvement regarding the testing speed.

Reply:
Created a jira for the changelog https://issues.apache.org/jira/browse/ARROW-13142